src/minirest_handler.erl

```diff
@@ -99,12 +99,27 @@ do_authorize(_Request, _Handler) ->
 do_parse_params(Request) ->
     Params = #{
         bindings => cowboy_req:bindings(Request),
-        query_string => maps:from_list(cowboy_req:parse_qs(Request)),
+        query_string => parse_qs(Request),
         headers => cowboy_req:headers(Request),
         body => #{}
     },
     do_read_body(Request, Params).
 
+parse_qs(Request) ->
+    lists:foldl(
+      fun({K, V}, MapAcc) ->
+              maps:update_with(
+                K,
+                fun([_|_] = OldV) -> [V | OldV];
```

Comment:
As far as I remember (from my times being a "web developer" 😅), the common convention understood by many frameworks is to add `[]` when we want an array, that is, converting `a[]=1&a[]=2&a[]=3` to `#{a => [1,2,3]}` (and
`a[b]=1&a[c]=2` to `#{a  => #{b => 1, c => 2}}`).

Reply:
Current implementation actually corresponds to [the convention in OpenAPI 3](https://spec.openapis.org/oas/v3.1.0#style-examples) with style = form and explode = true.

That said, if the plan is to rely on API spec later, and for example taking only the last element if the respective spec says that the parameter is not actually an array, then it sounds fine I would say.

Reply:
> As far as I remember (from my times being a "web developer" 😅), the common convention understood by many frameworks is to add `[]` when we want an array, that is, converting `a[]=1&a[]=2&a[]=3` to `#{a => [1,2,3]}` (and `a[b]=1&a[c]=2` to `#{a => #{b => 1, c => 2}}`).

this is mind blowing

Reply:
I think repeating QS params  (without `[]`) to represent lists is perfectly fine. My only concern is backward compatibility, which can be solved either in EMQX or probably in minirest (but we would need to pass an option to parse repeated params to lists).

Reply:
Agree

Reply:
That's how I propose to solve backward compatibility: https://github.com/SergeTupchiy/emqx/blob/EMQX-11900-multiple-clientid-username-clients-API/apps/emqx_dashboard/src/emqx_dashboard_swagger.erl#L463-L520

Drawbacks: 
 - maybe a bit complex/over-engineered (for a quite simple task)

Advantages:
 - it relies on hocon types to decide on how to wrap/unwrap Qs params, no need to introduce any new meta field to differentiate arrays from single QS params.
 - developers can continue defining QS params schemas as before or they can simply add `hocon:array(_)`:
 ```
hoconsc:mk(hoconsc:array(binary()), #{
    in => query,
    required => false,
    desc => <<"Client ID">>
})}
```
- can handle more complex types like `union(array(Type), SingleType)`
- respects custom converters, so that it's possible to define an array with a custom encoding, e. g. to parse coma separated lists